File: macos/.config/skhd/launcher/other/c/menu.c
#include "util_func.h"

#include <ctype.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

void app_menu()
{
    static const struct App apps[] = {
        {.key = 'b', .app = "Brave Browser", .icon = "  "},
        {.key = 'c', .app = "Visual Studio Code", .icon = " ﬏ "},
        {.key = 'd', .app = "Discord", .icon = " ﭮ "},
        {.key = 'k', .app = "KeePassXC", .icon = "  "},
        {.key = 'm', .app = "Messenger", .icon = "  "},
        {.key = 'o', .app = "Obsidian", .icon = "  "},
        {.key = 'p', .app = "Parallels Desktop", .icon = "  "},
        {.key = 's', .app = "System Preferences", .icon = "  "},
        {.key = 't', .app = "Microsoft Teams", .icon = "  "},
        {.key = 'v', .app = "VMware Fusion", .icon = "  "}};
    const int apps_n = sizeof(apps) / sizeof(apps[0]);
    char app_name[55], cmd[70], nl, key;
    int i;
    // char app_name[55], nl, key;

    for (i = 0; i < apps_n; i++) {
        if (i % 2)
            nl = '\n';
        else
            nl = '\0';
        strcpy(app_name, apps[i].app);
        // strcat(app_name, " ");
        strcat(app_name, apps[i].icon);
        printf("    [%c] %-25s%c", toupper(apps[i].key), app_name, nl);
    }

    key = gkey();
    i = key_apps(key, apps, apps_n);
    strcpy(cmd, "open -a \"");
    strcat(cmd, apps[i].app);
    strcat(cmd, "\"");
    system(cmd);
}

int main()
{
    app_menu();
    return 0;
}

File: macos/.config/skhd/launcher/other/python/menu.py
#!/usr/bin/env python
import subprocess
import os

Apps = {
    "b": { "app": "Brave Browser", "icon": " " },
    "c": { "app": "Visual Studio Code", "icon": "﬏ " },
    "d": { "app": "Discord", "icon": "ﭮ " },
    "k": { "app": "KeePassXC", "icon": " " },
    "m": { "app": "Messenger", "icon": " " },
    "o": { "app": "Obsidian", "icon": " " },
    "p": { "app": "Parallels Desktop", "icon": " " },
    "s": { "app": "System Preferences", "icon": " " },
    "t": { "app": "Microsoft Teams", "icon": " " },
    "v": { "app": "VMware Fusion", "icon": "" }
}

print()

i = 0
for k in Apps:
    if i % 2:
        nl='\n'
    else:
        nl=''
    i+=1
    print('    [{}] {:<25s}{}'.format(k.upper(), Apps[k]["app"] + ' ' + Apps[k]["icon"], nl), end='')


try:
    key = subprocess.check_output(os.path.expanduser('~') + '/.local/bin/readkey.sh', timeout=5).decode('utf-8').rstrip()
except subprocess.TimeoutExpired:
    exit()

if key in Apps:
    os.system(f'open -a "{Apps[key]["app"]}"')
else:
    exit()

